src/js/contentscripts/socialwidgets.js

```diff
@@ -229,9 +234,7 @@ function reinitializeWidgetAndUnblockTracker(button, urls, widgetElement) {
     // executed for buttons that have already been removed; we are trying
     // to prevent replacing an already removed button
     if (button.parentNode !== null) {
-      var codeContainer = document.createElement("div");
-      codeContainer.innerHTML = widgetElement.outerHTML;
-      button.parentNode.replaceChild(codeContainer, button);
+      button.parentNode.parentNode.parentNode.innerHTML = widgetElement.outerHTML;
```

Comment:
What's up with this line? Why go so many levels up? I think this is causing it to replace context along with the button -- see https://cyphe.rs/test.html

Reply:
Ah you found a bug (also I should've explained this line with a comment). We go up three levels to go from the image that was clicked to its bounding div to its bounding div ([this stuff](https://github.com/EFForg/privacybadger/blob/24ca14d3479d80c1043c40e043e21c4da9e65f1a/src/js/contentscripts/socialwidgets.js#L294-L331)). If you look at the examples I provided, this works out fine. I think what's different with your test page is that the parent of the parent of the parent happens to be the body element, not some otherwise-empty wrapper element that's OK to replace.

Reply:
This is a bit of an edge case though, please don't wait for the fix to keep reviewing.

Reply:
Thanks, I'll keep reviewing.

Reply:
OK, should be fixed.